app/models/group.rb

```diff
@@ -10,18 +10,17 @@
 class Group < ApplicationRecord
   belongs_to :leader, class_name: 'User'
   belongs_to :draw
-  has_many :memberships
+  has_many :memberships, dependent: :delete_all
```

Comment:
Note the use of `delete_all` due to callback loops when `destroy` was used (group is being destroyed, it calls destroy on the membership, which attempts to update the group and it gets messy and broken).